src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -662,12 +656,19 @@ internal static (UIElement? element, Branch? stale) SearchDownForTopMostElementA
 			// We didn't find any child at the given position, validate that element can be touched,
 			// and the position is in actual bounds(which might be different than the clipping bounds)
 #if __SKIA__
-			if (element.HitTest(transformToElement.Inverse().Transform(testPosition)) && renderingBounds.Contains(testPosition))
+			if (element.HitTest(transformToElement.Inverse().Transform(testPosition)))
 #else
-
 			if (elementHitTestVisibility == HitTestability.Visible && renderingBounds.Contains(testPosition))
 #endif
 			{
+				// UIElement.HitTest should be a stronger condition than elementHitTestVisibility == HitTestability.Visible
```

Comment:
UIElement.HitTest just does Visual hit testing, which isn't aware of whether the UIElement's background is NULL or not. Aren't we creating transparent background in both cases? Also, what if the user gets the Visual and modifies the background, in this case they are not the same. I tend to think we should really restore `elementHitTestVisibility == HitTestability.Visible` check

Reply:
Not sure if it's the case so far, but the `Visual` **could** be aware if ~the background is `null`~ a pixel has not been painted, or not if it has been painted using with transparent color. If so, the `elementHitTestVisibility == HitTestability.Visible` check would indeed only be about what hit-testability we push to the native element and therefor should be less restrictive than the `IsViewHit()`.